File: VB/Controllers/HomeController.vb
Imports System
Imports System.Web.Mvc
Imports Example.Models
Imports DevExpress.Web.Mvc

Namespace Example.Controllers

    <HandleError>
    Public Class HomeController
        Inherits Controller

        Public Function Index() As ActionResult
            ViewData("Message") = "Welcome to DevExpress Extensions for ASP.NET MVC!"
            Dim defaultOrder As Order = New Order() With {.ProductName = "Product", .Price = 12.3D, .Customer = New Customer() With {.FullName = "Vest", .BirthDate = New DateTime(1892, 1, 3)}}
            Return View(defaultOrder)
        End Function

        <HttpPost>
        Public Function PostOrder(<ModelBinder(GetType(DevExpressEditorsBinder))> ByVal order As Order) As ActionResult
            ViewData("Message") = order.Customer.FullName & "'s order has been posted"
            Return View("Index", order)
        End Function
    End Class
End Namespace

File: CS/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using Example.Models;
using DevExpress.Web.Mvc;

namespace Example.Controllers {
    [HandleError]
    public class HomeController : Controller {
        public ActionResult Index() {
            ViewData["Message"] = "Welcome to DevExpress Extensions for ASP.NET MVC!";

            Order defaultOrder = new Order() {
                ProductName = "Product",
                Price = 12.3M,
                Customer = new Customer() {
                    FullName = "Vest",
                    BirthDate = new DateTime(1892, 1, 3)
                }
            };

            return View(defaultOrder);
        }

        [HttpPost]
        public ActionResult PostOrder([ModelBinder(typeof(DevExpressEditorsBinder))] Order order) {
            ViewData["Message"] = order.Customer.FullName + "'s order has been posted";

            return View("Index", order);
        }
    }
}
